### DISCUSSION TITLE
Grep specific modules under node_modules directory

### DISCUSSION THREAD

**Discussion by dbarros** [nvim-neo-tree/neo-tree.nvim#1428] [Q&A] [answered]
When doing a grep across a React project, I'd like to add a couple of modules in the node_modules directory so that they are also searched (along with the project itself). Is this possible? This would be only applicable to one project, not every project I open in Neovim.

**Comment by pysan3** [comment]
`find_args` may be a function so you can specify different args in some circumstances. 

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/660427550bcb75ff161e65695190f7760ae951e0/lua/neo-tree/defaults.lua#L491

However scanning more files means more overhead so be careful or your neovim may freeze for some amount of time.

**Reply by dbarros** [reply]
Thank you, @pysan3. Instead of changing the global Neotree config, is there a way to have a config file at the project-level? So if I was to change the options for the `--find_comand`, it would apply to a specific project, but if I was editing anything else (other than that one project), the default config would continue to be used.

**Reply by pysan3** [reply, marked as answer]
@dbarros `find_args` can be a function that will be called on the fly as it is explained in defaults.lua. 

You can use whatever lua method to detect the current project and branch out depending on the condition. 

`vim.fn.getcwd() == "/path/to/my-project"` or anything else. 

If you are not familiar with lua, tell me what you want and I can help you.
